R/sparklyr-spark-connect.R

```diff
@@ -102,34 +109,61 @@ spark_connect_method.spark_method_databricks_connect <- function(
     return(invisible)
   }
 
-  db <- import_check("databricks.connect", envname, silent)
+  dbc <- import_check("databricks.connect", envname, silent)
+  db_sdk <- import_check("databricks.sdk", envname, silent = TRUE)
 
   if (!is.null(cluster_info)) {
     msg <- "{.header Connecting to} {.emph '{cluster_info$name}'}"
     msg_done <- "{.header Connected to:} {.emph '{cluster_info$name}'}"
     master_label <- glue("{cluster_info$name} ({cluster_id})")
-  } else {
+  } else if (!serverless) {
     msg <- "{.header Connecting to} {.emph '{cluster_id}'}"
     msg_done <- "{.header Connected to:} '{.emph '{cluster_id}'}'"
     master_label <- glue("Databricks Connect - Cluster: {cluster_id}")
+  } else if (serverless) {
+    msg <- "{.header Connecting to} {.emph 'serverless'}"
+    msg_done <- "{.header Connected to:} '{.emph 'serverless'}'"
+    master_label <- glue("Databricks Connect - Cluster: serverless")
   }
 
   if (!silent) {
     cli_div(theme = cli_colors())
     cli_progress_step(msg, msg_done)
   }
 
-  remote_args <- list()
-  if (master != "") remote_args$host <- master
-  if (token != "") remote_args$token <- token
-  if (cluster_id != "") remote_args$cluster_id <- cluster_id
+  # sdk config
+  conf_args <- list(host = master)
+  # if token is found, propagate
+  # otherwise trust in sdk to detect and do what it can?
+  if (token != "") {
+    conf_args$token <- token
+    conf_args$auth_type <- "pat"
+  }
 
-  databricks_session <- function(...) {
-    user_agent <- build_user_agent()
-    db$DatabricksSession$builder$remote(...)$userAgent(user_agent)
+  if (serverless) {
+    conf_args$serverless_compute_id <- "auto"
+  } else {
+    conf_args$cluster_id <- cluster_id
   }
 
-  conn <- exec(databricks_session, !!!remote_args)
+  sdk_config <- db_sdk$core$Config(!!!conf_args)
+
+  # unsure if this iss needed anymore?
```

Comment:
I think we need to remove this from here, specially since we can't use `httr2:::is_hosted_session()` since a `:::` is not allowed. Do you think this is important for the package to do if the user is on desktop? If so, what do you think about isolating it in its own exported function? Maybe `pysparklyr::databricks_desktop_login()`?

Reply:
I don't think this is required, will do some testing without.